pyFAST/fastfarm/FASTFarmCaseCreation.py

```diff
@@ -2022,15 +2194,397 @@ def FF_slurm_submit(self, qos='normal', A=None, t=None, delay=4):
+        for cond in range(self.nConditions):
+            for case in range(self.nCases):
+                for seed in range(self.nSeeds):
+                    seedPath = os.path.join(self.path, self.condDirList[cond], self.caseDirList[case], f'Seed_{seed}')
+        
+                    # Recover case properties
+                    D_       = self.allCases['D'   ].max().values # Getting the maximum in case different turbines are present
+                    HubHt_   = self.allCases['zhub'].max().values # Getting the maximum in case different turbines are present
+                    Vhub_    = self.allCond.sel(cond=cond)['vhub'   ].values
+                    shear_   = self.allCond.sel(cond=cond)['shear'  ].values
+                    tivalue_ = self.allCond.sel(cond=cond)['TIvalue'].values
+                    # Recover turbine properties (array of length nTurbines)
+                    xWT = self.allCases.sel(case=case)['Tx'].values
+                    yWT = self.allCases.sel(case=case)['Ty'].values
+                    zWT = self.allCases.sel(case=case)['Tz'].values 
+        
+        
+                    # --------------- FAST.Farm ----------------- #
+                    templateFSTF = os.path.join(self.templatePath, self.FFfilename)
+                    outputFSTF   = os.path.join(seedPath, 'FFarm_mod.fstf')
+        
+                    path=os.path.dirname(self.path)                     
+                    WindPath = os.path.join(path, 'Wind')
+                    MannPath = os.path.join(WindPath, 'MannBox')
+                    FarmPath = os.path.join(MannPath, f'Farm_{len(self.wts)}T')
+                    WindirPath = os.path.join(FarmPath, f"Windir_{f'{self.inflow_deg[case]:+03d}'.replace('+','p').replace('-','m')}")
+                    Path = os.path.join(WindirPath, f'Seed_{seed}')
+                    MBfilePath = Path
+                    MannFile =  f'Vhub{Vhub_}_TI{tivalue_}_PL{shear_}_Time{self.tmax}s'
+                    f=open(os.path.join(MBfilePath,MannFile+'_u.txt'),'r')
+                    lines=f.read().splitlines()
+                    nx=None; dx=None; dy=None; ny=None; dz=None; nz=None
+                    for i in range(len(lines)):
+                        if nx == None:
+                            nx=re.search('Nx :(\d+)', lines[i])
+                        if dx == None:
+                            dx=re.search('Dx : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])
+                        if ny == None:
+                            ny=re.search('Ny :(\d+)', lines[i])
+                        if dy == None:
+                            dy=re.search('Dy : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])
+                        if nz == None:
+                            nz=re.search('Nz :(\d+)', lines[i])
+                        if dz == None:
+                            dz=re.search('Dz : ((?:0|[1-9]\d*)(\.\d*)?(?:[eE][+\-]?\d+))', lines[i])               
```

Comment:
Regex like this should come with some comments above. It is hard to tell what is being done here.

Reply:
I am reading the parameters of the MannBox as they have to be identical to the parameters in the Inflowind.dat file